emulator/src/main/java/com/scalar/client/tool/emulator/EmulatorTerminal.java

```diff
@@ -211,4 +214,33 @@ private boolean parseAndRunCommand(String line) {
     }
     return false;
   }
+
+  private String[] paramsParser(String params) {
+    Stack<String> stack = new Stack<String>();
+    List<String> paramsList = new ArrayList<String>();
+    ByteArrayOutputStream byteArray = new ByteArrayOutputStream();
+
+    for (int i = 0; i < params.length(); i++) {
+      String character = Character.valueOf(params.charAt(i)).toString();
+      if (character.matches("[{\"}]")) {
+        if (!stack.empty()) {
+          if ((character.equals("\"") && stack.peek().equals("\""))
+              || (character.equals("}") && stack.peek().equals("{"))) {
+            stack.pop();
+          } else {
+            stack.push(character);
+          }
+        } else {
+          stack.push(character);
+        }
+      }
+
+      byteArray.write(params.charAt(i));
+      if ((character.matches(" ") && stack.empty()) || i == params.length() - 1) {
+        paramsList.add(byteArray.toString().trim());
+        byteArray.reset();
+      }
+    }
+    return paramsList.toArray(new String[0]);
```

Comment:
Basically this is a stack and queues implementation...
1. Each character of input will be streamed via `param.charAt(i)`
2. Whenever a special character (`"` or `{` or `}`) is detected, it will be pushed into the stack.
3. ONLY pop IF `"` detected and last pushed character is `"`. OR IF `}` detected and last pushed character `{`.
4 `byteArray` acts as a buffer, it will take in characters until `(character.matches(" ") && stack.empty()) || i == params.length() - 1)` condition is met...if so, its content will be added to the paramsList and then `.reset()`.